save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/internal/TestSuitesSourceInternalController.kt

```diff
@@ -102,32 +97,40 @@ class TestSuitesSourceInternalController(
 
     private fun TestSuitesSourceDto.downloadSnapshot(
         version: String
-    ): Mono<ByteBufferFluxResponse> = testSuitesSourceVersionService.doesContain(organizationName, name, version)
+    ): Mono<ByteBufferFluxResponse> = testsSourceVersionService.doesContain(organizationName, name, version)
         .filter { it }
         .switchIfEmptyToNotFound {
             "Not found a snapshot of $name in $organizationName with version=$version"
         }
         .map {
             ResponseEntity.ok()
                 .contentType(MediaType.APPLICATION_OCTET_STREAM)
-                .body(testSuitesSourceVersionService.download(organizationName, name, version))
+                .body(testsSourceVersionService.download(organizationName, name, version))
         }
 
-    companion object {
-        private val log: Logger = getLogger<TestSuitesSourceService>()
+    private fun TestsSourceSnapshotDto.toKey() = testSuitesSourceRepository.getByIdOrNotFound(sourceId)
+        .let { source ->
+            TestSuitesSourceSnapshotKey(
+                organizationName = source.organization.name,
+                testSuitesSourceName = source.name,
+                version = commitId,
+                creationTime = commitTime,
+            )
+        }
 
-        private fun TestsSourceSnapshotInfo.toKey() = TestSuitesSourceSnapshotKey(
+    companion object {
+        private fun TestsSourceVersionInfo.toKeyForSnapshot() = TestSuitesSourceSnapshotKey(
```

Comment:
just to clarify, we will remove `TestSuitesSourceSnapshotKey` in future, right? as been said in https://github.com/saveourtool/save-cloud/pull/1727#discussion_r1072081276

Reply:
yes, I want to leave it as long as possible during refactoring to move all public api changes on late stage